tiledb/sm/misc/types.h

```diff
@@ -134,10 +134,13 @@ typedef std::vector<Range> NDRange;
 /** A value as a vector of bytes. */
 typedef std::vector<uint8_t> ByteVecValue;
 
-/** Contains the buffer(s) and buffer size(s) for some attribute / dimension. */
+/** A byte vector. */
+typedef std::vector<uint8_t> ByteVec;
```

Comment:
Since we have `ByteVecValue` as a binary representation of an arbitrary type values, is it possible that the semantics of this `ByteVec` could be described as a representation of a _list_ of arbitrary type values? If so, let's rename this to `ByteVecValueList`. If this is just a generic binary payload, then let's leave this as-is.

Reply:
Yes, for now I suggest we keep it generic, but if we start using it more as a generic value, we will rename. Initially I had a name similar to what you suggest, but removed it as it wasn't used that way a lot in the code. An easy change down the road if need be.